modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -636,10 +636,12 @@ private void updateComponentSize() {
             }
             scaleFactorX = newScaleFactorX;
             scaleFactorY = newScaleFactorY;
-            if (stage != null) {
-               stage.setRenderScaleX(scaleFactorX);
-               stage.setRenderScaleY(scaleFactorY);
-            }
+            Platform.runLater(() -> {
+                if (stage != null) {
+                   stage.setRenderScaleX(scaleFactorX);
+                   stage.setRenderScaleY(scaleFactorY);
+                }
+            });
```

Comment:
[question]
could there be race condition between sendResizeEventToFX() and stage.setRenderScale() since we now have two different threads operating?

Reply:
I dont think so as we have similar call on FX thread in setScene() without any race issue...